test/integrations/destinations/sfmc/processor/data.ts

```diff
@@ -1894,4 +1894,328 @@ export const data = [
+                    from: 'message event',
+                    to: 'test-event-definition',
+                  },
+                ],
+                externalKey: 'f3ffa19b-e0b3-4967-829f-549b781080e6',
+                subDomain: 'testHandleHttpRequest401',
+              },
+              Enabled: true,
+              Transformations: [],
+            },
+          },
+        ],
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: [
+          {
+            error:
+              '{"message":"Could not retrieve access token","destinationResponse":"{\\"response\\":{\\"error\\":\\"invalid_client\\",\\"error_description\\":\\"Invalid client ID. Use the client ID in Marketing Cloud Installed Packages.\\",\\"error_uri\\":\\"https://developer.salesforce.com/docs\\"},\\"status\\":401}"}',
+            statTags: {
+              destType: 'SFMC',
+              errorCategory: 'network',
+              errorType: 'aborted',
+              feature: 'processor',
+              implementation: 'native',
+              module: 'destination',
+            },
+            statusCode: 401,
+          },
+        ],
+      },
+    },
+  },
+  {
+    name: 'sfmc',
+    description: 'Tests 429 status code from sfmc',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            message: {
+              event: 'message event',
+              type: 'track',
+              userId: '12345',
+              properties: {
+                id: 'id101',
+                contactId: 'cid101',
+                email: 'testemail@gmail.com',
+                accountNumber: '99110099',
+                patronName: 'SP',
+              },
+            },
+            destination: {
+              ID: '1pYpzzvcn7AQ2W9GGIAZSsN6Mfq',
+              Name: 'SFMC',
+              DestinationDefinition: {
+                ID: '1pYpYSeQd8OeN6xPdw6VGDzqUd1',
+                Name: 'SFMC',
+                DisplayName: 'Salesforce Marketing Cloud',
+                Config: {
+                  destConfig: [],
+                  excludeKeys: [],
+                  includeKeys: [],
+                  saveDestinationResponse: false,
+                  supportedSourceTypes: [],
+                  transformAt: 'processor',
+                },
+                ResponseRules: {},
+              },
+              Config: {
+                clientId: 'testHandleHttpRequest429',
+                clientSecret: 'testHandleHttpRequest429',
+                subDomain: 'testHandleHttpRequest429',
+              },
+              Enabled: true,
+              Transformations: [],
+            },
+          },
+        ],
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: [
+          {
+            error:
+              '{"message":"Could not retrieve access token","destinationResponse":"Your requests are temporarily blocked."}',
```

Comment:
Interesting that `destinationResponse` only contains the error message(string) sent by destination as opposed to the whole response body.
Might be more useful to have response body

Reply:
updated , not printing whole response body